### PR TITLE
Add Support for Travis CI And Rubocop

### PR DESCRIPTION
**Pull request by mrfishball**
This change enables Rubocop for code linting/formatting as well as enabling Travis CI to run unit test against the Ruby version defined in the `.ruby-version` file.

**Changes:**
- Added `.travis.yml`, `.rubocop.yml`
- Updated `Gemfile` and `Gemfile.lock`
- Update code formatting for varies files
- Updated README

### CODE REVIEW

File: config/application.rb [hunk 1]
```diff
@@ -1,3 +1,5 @@
+# frozen_string_literal: true
```
Comment (MollieS) [inline config/application.rb:1]: Can we get rid of this rule in rubocop and remove these comments?

File: .ruby-version [hunk 2]
```diff
@@ -1 +1 @@
-ruby-2.5.5
+ruby-2.6.5
```
Comment (MollieS) [inline .ruby-version:1]: Is there a reason we're not using the latest ruby version? 2.7.X

File: Gemfile [hunk 3]
```diff
@@ -1,32 +1,32 @@
...
-git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 
 ruby '2.6.5'
 
-gem 'rails', '~> 6.0.2', '>= 6.0.2.1'
-gem 'sqlite3', '~> 1.4'
-gem 'puma', '~> 4.1'
 gem 'bootsnap', '>= 1.4.2', require: false
 gem 'devise_token_auth'
+gem 'puma', '~> 4.1'
```
Comment (MollieS) [inline Gemfile:9]: Can we use rails server instead of puma? Do we need to pull puma in?

File: config/application.rb (hunk 1 above)
Comment (mrfishball) [reply config/application.rb:1]: Is there a reason why we want to get rid of this rule completely? We can alternatively exclude certain files from this rule. What do you think?

File: .ruby-version (hunk 2 above)
Comment (mrfishball) [reply .ruby-version:1]: I noticed the version declared in the `ruby-version` file didn't match that of the Gemfile so I decided to go with one in the Gemfile instead. I don't see why we can't use the latest version though. Will update that.

File: Gemfile (hunk 3 above)
Comment (mrfishball) [reply Gemfile:9]: My understanding is that `Puma` is the default web server for Rails. When we call `bundle exec rails server` Rails runs Puma.

File: .ruby-version (hunk 2 above)
Comment (mrfishball) [reply .ruby-version]: Bumped Ruby version to 2.7.1. https://github.com/SteelWarriors/steel-warriors-api/pull/3/commits/4fd8b12930e13ff96062322e838eaf389512f57a

File: config/application.rb (hunk 1 above)
Comment (mrfishball) [reply config/application.rb:1]: Resolving this for now. Let's revisit this another time.

**Comment by mrfishball** [comment]
Merged this branch. I think as development starts rolling, CI and Rubocop will be crucial for keeping us in check. @MollieS As I mentioned above, let's revisit some of the comments that we didn't get to address and make changes accordingly!

**Pull request merged**